core/sdk/src/main/java/com/mongodb/stitch/core/auth/internal/CoreStitchAuth.java

```diff
@@ -196,6 +197,27 @@ public Response doAuthenticatedRequest(final StitchAuthRequest stitchReq) {
     }
   }
 
+  @Override
+  public <T> Stream<T> openAuthenticatedStream(final StitchRequest stitchReq,
+                                               final Decoder<T> decoder) {
+    return openAuthenticatedStream(stitchReq, decoder, true);
+  }
+
+  private <T> Stream<T> openAuthenticatedStream(final StitchRequest stitchReq,
+                                                final Decoder<T> decoder,
+                                                boolean tryRefresh) {
+    try {
+      return new Stream<>(
+          requestClient.doStreamRequest(stitchReq.builder().withPath(
+              stitchReq.getPath() + AuthStreamFields.AUTH_TOKEN + getAuthInfo().getAccessToken()
+          ).build()),
+          decoder
+      );
+    } catch (final StitchServiceException ex) {
+      return handleAuthFailure(ex, stitchReq, decoder, tryRefresh);
```

Comment:
This isn't actually handling errors still. It needs to be similar to handleRequestError meaning you may need to inspect the event for an error.

Reply:
Try to reuse the code that exists for this case.

Reply:
👍